mobile/examples/phi-3/android/app/src/main/java/ai/onnxruntime/genai/demo/GenAIWrapper.java

```diff
@@ -0,0 +1,61 @@
+package ai.onnxruntime.genai.demo;
+
+import android.util.Log;
+
+public class GenAIWrapper implements AutoCloseable {
+    // Load the GenAI library on application startup.
+    static {
+        System.loadLibrary("genai"); // JNI layer
+        System.loadLibrary("onnxruntime-genai");
+        System.loadLibrary("onnxruntime");
+    }
+
+    private final long nativeModel;
+    private final long nativeTokenizer;
+    private TokenUpdateListener listener;
+
+    public interface TokenUpdateListener {
+        void onTokenUpdate(String token);
+    }
+
+    public GenAIWrapper(String modelPath) {
```

Comment:
```suggestion
    public GenAIWrapper(String modelPath) throws GenAIException {
```

The throw is from the C++ code, but should probably be explicitly declared here.